conjure-go-client/httpclient/internal/course_exponential_decay_reservoir.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2021 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"math"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+const (
+	decaysPerHalfLife = 10
+)
+
+var (
+	decayFactor = math.Pow(0.5, 1.0/decaysPerHalfLife)
+)
+
+type CourseExponentialDecayReservoir interface {
+	Update(updates float64)
+	Get() float64
+}
+
+var _ CourseExponentialDecayReservoir = (*reservoir)(nil)
+
+type reservoir struct {
+	value                    float64
+	lastDecay                int64
+	nanoClock                func() int64
+	decayIntervalNanoseconds int64
+	mu                       sync.Mutex
```

Comment:
Can you re-order this struct so the mutex sits on top of the value it's protecting, which appears to just be `value` ([docs](https://dmitri.shuralyov.com/idiomatic-go/entries/2))